openpype/hosts/maya/api/lib_rendersettings.py

```diff
@@ -300,3 +306,79 @@ def _additional_attribs_setter(self, additional_attribs):
                         attribute=attribute,
                         attribute_type=attribute_type)
                 )
+
+    def _set_3delight_settings(self, width, height):
+        """Sets important settings for 3Delight. """
+
+        # resolution
+        cmds.setAttr("defaultResolution.width", width)
+        cmds.setAttr("defaultResolution.height", height)
+        cmds.setAttr("defaultRenderGlobals.animation", 1)
+
+        # frame range
+        start_frame = int(cmds.playbackOptions(query=True,
+                                               animationStartTime=True))
+        end_frame = int(cmds.playbackOptions(query=True,
+                                             animationEndTime=True))
+
+        dl_render_settings = cmds.ls(type="dlRenderSettings")
+        assert len(dl_render_settings) >= 1
+        for dl_render_setting in dl_render_settings:
+            cmds.setAttr(
+                "{}.startFrame".format(dl_render_setting), start_frame)
+            cmds.setAttr(
+                "{}.endFrame".format(dl_render_setting), end_frame)
+
+            # outputOptionsDefault
+            cmds.setAttr(
+                "{}.outputOptionsDefault".format(dl_render_setting), 2)
+
+        """
+        3delight doesn't use maya's render layers to indicate what to render,
+        instead, it uses dlRenderSettings as specified by the connection to
+        dlRenderGlobals1, of which there is only one. Since this is not the
+        "mayaonic" way, we must insert a pre-render MEL into maya's
+        "defaultRenderGlobals.preMel", which will help us work this out. We
+        wouldn't this this ... "hack" if deadline allowed us to insert commands
+        into its MayaBatch stuff, or, if it allowed setting up renderers in a
+        sensible way.
+        """
+
+        # We need the DOTALL since the regex will now match across newlines.
+        regex_match = re.compile(r"\/\*--\*3dl_v\d+\*--\*\/.*\/\*--\*\*--\*\/",
+                                 re.DOTALL)
+        PREMEL_TEMPLATE = """/*--*3dl_v4*--*/
+currentTime -e `getAttr ("defaultRenderGlobals.startFrame")`;
```

Comment:
Why do we need to set the current time to the start frame as a pre-render script? Also, doesn't this logic enforce that it'll always render the full frame range somehow?

I feel like this particular pre-render snippet is lacking some comments to explain what the code itself is doing to resolve the issue at hand.

The lengthy comment above does tell us that there's something to work around, but it's not telling us how it's solving the problem at hand. From just reading the comment and reading the mel script I have still no idea why this works the way it does and how it solves what it intends.

I think we should clarify this code for future maintainers.